WireAPI/Tests/WireAPITests/APIs/FeatureConfigsAPI/FeatureConfigsAPITests.swift

```diff
@@ -139,60 +142,45 @@ final class FeatureConfigsAPITests: XCTestCase {
 
     // MARK: - V4 & V5
 
-    func testGetFeatureConfigs_SuccessResponse_200_V4_TO_V5() async throws {
+    func testGetFeatureConfigs_SuccessResponse_200_V4_To_V5_Then_Verify_Requests() async throws {
         // Given
-        let httpClient = try HTTPClientMock(code: .ok, payloadResourceName: "GetFeatureConfigsSuccessResponseV4")
+        let apiService = MockAPIServiceProtocol.withResponses([
+            (.ok, "GetFeatureConfigsSuccessResponseV4")
+        ])
 
         let supportedVersions: [APIVersion] = [.v4, .v5]
 
-        let suts = supportedVersions.map { $0.buildAPI(client: httpClient) }
-
-        try await withThrowingTaskGroup(of: [FeatureConfig].self) { taskGroup in
-            for sut in suts {
-                taskGroup.addTask {
-                    // When
-                    try await sut.getFeatureConfigs()
-                }
-
-                for try await result in taskGroup {
-                    // Then
-                    XCTAssertEqual(
-                        result,
-                        Scaffolding.featureConfigsV4
-                    )
-                }
-            }
+        // Then
+        try await apiSnapshotHelper.verifyRequest(for: supportedVersions, apiService: apiService) { sut in
```

Comment:
These tests are adding more snapshots when asserting the response handling, is that intentional?

Reply:
yes, because we want to assert the model response but also ensure the request is correct for a given response. let's say the request changed for a specific version, without it we wouldn't know something changed, no?